log/src/io/pedestal/log.clj

```diff
@@ -388,23 +388,25 @@
   Note:
   If you mix `with-context` with the more basic `with-context-kv`, you may see undesired keys/values in the log"
   [ctx-map & body]
-  (let [formatter (::formatter ctx-map pr-str)]
-    (if (empty? ctx-map) ;; Optimize for the code-gen/dynamic case where the map may be empty
-      `(do
-         ~@body)
-      `(let [old-ctx# *mdc-context*
-             mdc# (or ~(::mdc ctx-map) (MDC/getMDCAdapter))]
-         (binding [*mdc-context* (merge *mdc-context* ~ctx-map)]
-           (-put-mdc mdc# mdc-context-key (~formatter (dissoc *mdc-context*
-                                                              :io.pedestal.log/formatter
-                                                              :io.pedestal.log/mdc)))
-           (try
-             ~@body
-             (finally
-               (-put-mdc mdc# mdc-context-key ((:io.pedestal.log/formatter old-ctx# pr-str)
-                                               (dissoc old-ctx#
-                                                       :io.pedestal.log/formatter
-                                                       :io.pedestal.log/mdc))))))))))
+  (if (or (symbol? ctx-map)
```

Comment:
Limiting to just symbol here feels too constraining, as I could easily imagine wanting to invoke a function to generate the context map, rather than it being inline only.  As currently coded, if an expression is provided, it will be treated as an empty map, which will certainly confuse users.

Reply:
Do you mean an expression like
```clojure
(log/with-context ((constantly {:a :b})) ["some" 'body])
```
or a var pointing to a function?:
```clojure
(let [gen-ctx-map-fn (constantly {:a :b})]
  (log/with-context gen-ctx-map-fn ["some" 'body]))
```

For the second it isn't possible (at macro expansion time) to know what is a function that the macro should expand into code that invokes it, and what is pointing to a base map. Doing it at eval time would work but go against the [suggestion to limit overhead](https://github.com/pedestal/pedestal/pull/713#discussion_r867978508).

For the first example, it would be possible to detect the expression at expansion time and have the macro emit code that shifts the work, that is, the evaluation of `(::formatter ctx-map pr-str)` and `(::mdc ctx-map)` expressions to normal runtime

Reply:
```
(log/with-context (create-context-from-request request) ....
```

The `(create-context-from-request ...)` form would fail the symbol? check.

Reply:
my comment was a bit confusing, so now I'm even more confused. You thinking something like this https://github.com/pedestal/pedestal/pull/713/commits/180332087742b91f4abbffb025a7c890c6868d77 ?